frontend/src/metabase/public/containers/PublicOrEmbeddedDashboard/PublicOrEmbeddedDashboard.tsx

```diff
@@ -98,20 +99,22 @@ class PublicOrEmbeddedDashboardInner extends Component<PublicOrEmbeddedDashboard
       dashboardId,
     } = this.props;
 
-    initialize();
+    if (!dashboard) {
```

Comment:
I think this block has a problem, I'm trying to trace the code, but from my testing, if you go to overview page in Shoppy and select other dashboard, only the first dashboard that was loaded will be shown
![image](https://github.com/metabase/metabase/assets/1937582/c8265451-db9e-4095-9ef1-458760f32c8e)

Yeah, I've tested this 2 times. I'm pretty sure this causes the problem. The reason is that `dashboard` is from redux, when we want to fetch a new dashboard the only thing that's change is `dashboardId` so this is where we call `_initialize` to fetch a new dashboard again
https://github.com/metabase/metabase/blob/bf7dc9829ec33d77afbeec89754e32d6da83f570/frontend/src/metabase/public/containers/PublicOrEmbeddedDashboard/PublicOrEmbeddedDashboard.tsx#L134-L137

if we have this if block, it will never fetch a new dashboard because `dashboard` here is already present, from a redux store, and the result is the old dashboard will be shown. You need to remove this if block.